client/my-sites/plans/jetpack-plans/product-card/index.tsx

```diff
@@ -236,8 +249,21 @@ const ProductCard: React.FC< ProductCardProps > = ( {
 			scrollCardIntoView={ scrollCardIntoView }
 			collapseFeaturesOnMobile={ collapseFeaturesOnMobile }
 			pricesAreFetching={ pricesAreFetching }
+			isPricingPageTreatment202204={ isPricingPageTreatment202204 }
+			isPricingPageTest202204Loading={ isLoadingExperimentAssignment }
+			belowButtonText={ isPricingPageTreatment202204 ? 'Renews at the normal rate.' : '' }
 		/>
 	);
 };
 
-export default ProductCard;
+export default connect( ( state ) => {
+	const siteId = getSelectedSiteId( state ) ?? 0;
```

Comment:
If there is no site selected in the calypso state, this will return 0, which causes `getCurrentRouteParameterized` to return null. If you use the site switcher in Calypso and click "Switch Site" and then "All my sites", this seems to unset the current selected site in the state.

I had trouble initially getting the treatment condition on the Jetpack cloud pricing page and the Jetpack connect store page because I had not selected a site in the site switcher yet. Maybe the route determination needs to vary if the site ID is or is not defined?

Reply:
> which causes getCurrentRouteParameterized to return null

Aha - great catch, thanks. I'll revise things to fix this.